java-frontend/src/main/java/org/sonar/java/se/xproc/MethodBehavior.java

```diff
@@ -84,11 +84,7 @@ public void createYield(ExplodedGraph.Node node, boolean storeNodeForReporting)
       ExceptionalYield exceptionalYield = new ExceptionalYield(nodeForYield, this);
       if (resultSV != null) {
         Type type = ((SymbolicValue.ExceptionalSymbolicValue) resultSV).exceptionType();
-        String typeName = null;
-        if(type != null) {
-          typeName = type.fullyQualifiedName();
-        }
-        exceptionalYield.setExceptionType(typeName);
+        exceptionalYield.setExceptionType(type);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in method 'setExceptionType' declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=1986&issues=AWKvagk8T11gSzukJmgW&open=AWKvagk8T11gSzukJmgW)

[ISSUE_KEY]: AWKvagk8T11gSzukJmgW